tests/test_sdeint.py

```diff
@@ -35,152 +36,149 @@
-            self.assertEqual(expected, None)
-        else:
-            torch.testing.assert_allclose(actual, expected, rtol=rtol, atol=atol)
 
 
-if __name__ == '__main__':
-    unittest.main()
+class _nullcontext:
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+
+@pytest.mark.parametrize('device', devices)
+def test_rename_methods(device):
+    # Test renaming works with a subset of names.
+    sde = basic_sde.CustomNamesSDE().to(device)
+    y0 = torch.ones(batch_size, d, device=device)
+    ts = torch.linspace(t0, t1, steps=T, device=device)
+    ans = torchsde.sdeint(sde, y0, ts, dt=dt, names={'drift': 'forward'})
+    assert ans.shape == (T, batch_size, d)
+
+
+@pytest.mark.skip("Temporarily deprecating logqp.")
+@pytest.mark.parametrize('device', devices)
+def test_rename_methods_logqp(device):
+    # Test renaming works with a subset of names when `logqp=True`.
+    sde = basic_sde.CustomNamesSDELogqp().to(device)
+    y0 = torch.ones(batch_size, d, device=device)
+    ts = torch.linspace(t0, t1, steps=T, device=device)
+    ans = torchsde.sdeint(sde, y0, ts, dt=dt, names={'drift': 'forward', 'prior_drift': 'w'}, logqp=True)
+    assert ans[0].shape == (T, batch_size, d)
+    assert ans[1].shape == (T - 1, batch_size)
+
+
+def _use_bm__levy_area_approximation():
+    yield False, None
+    yield True, 'none'
+    yield True, 'space-time'
+    yield True, 'davie'
+    yield True, 'foster'
+
+
+@pytest.mark.parametrize('sde_cls', [basic_sde.ScalarSDE, basic_sde.AdditiveSDE, basic_sde.DiagonalSDE,
+                                     basic_sde.GeneralSDE])
+@pytest.mark.parametrize('use_bm,levy_area_approximation', _use_bm__levy_area_approximation())
+@pytest.mark.parametrize('sde_type', ['ito', 'stratonovich'])
+@pytest.mark.parametrize('method', ['blah', 'euler', 'milstein', 'srk', 'euler_heun', 'heun', 'midpoint', 'log_ode'])
+@pytest.mark.parametrize('adaptive', [False, True])
+@pytest.mark.parametrize('logqp', [False, True])
+@pytest.mark.parametrize('device', devices)
+def test_sdeint_run_shape_method(sde_cls, use_bm, levy_area_approximation, sde_type, method, adaptive, logqp, device):
+    # Tests that sdeint:
+    # (a) runs/raises an error as appropriate
+    # (b) produces tensors of the right shape
+    # (c) accepts every method
+
+    if logqp:
+        pytest.skip("Temporarily deprecating logqp.")
+
+    should_fail = False
+    if sde_type == 'ito':
+        if method in ('euler_heun', 'heun', 'midpoint', 'log_ode'):
+            should_fail = True
+    else:
+        if method in ('euler', 'srk'):
+            should_fail = True
+    if method in ('milstein', 'srk') and sde_cls.noise_type == 'general':
+        should_fail = True
+    if method == 'srk' and levy_area_approximation == 'none':
+        should_fail = True
+    if method == 'log_ode' and levy_area_approximation in ('none', 'space-time'):
+        should_fail = True
+    if method == 'blah':
```

Comment:
I think we could save ourselves from checking `method=='blah'` if we did the not-in-method-list-checks above.

Reply:
Done.